tensorflow_estimator/python/estimator/estimator.py

```diff
@@ -1036,15 +1036,51 @@ def _validate_features_in_predict_input(self, result):
                       'QueueRunner. That means predict yields forever. '
                       'This is probably a mistake.')
 
-  def _get_iterator_from_input_fn(self, input_fn, mode, distribution=None):
-    if distribution is not None:
+  def _has_input_context_arg(self, input_fn_args):
+    # Only support 4 arguments at most. They are `mode`, `config`, `params`
```

Comment:
@karmel Is this fine if we limit args within ["mode", "config", "params",  "input_context"]?

Reply:
Actually, it seems strange  in `Estimator`. Because `input_fn` should not have args in `Estimator`. The document doesn't mentioned `config`, `params` and `mode` in `input_fn` and it will take effect only with these spelling. Does the users know this features ?

Reply:
In general, I would feel much more comfortable to adding input_context= to the set of kwargs we pass through; why does input_context have to be a positional arg, and the first one? Can we instead call with some established keyword so we can check and explicitly catch?